packages/dbml-parse/src/lib/analyzer/validator/elementValidators/note.ts

```diff
@@ -26,18 +27,43 @@ export default class NoteValidator implements ElementValidator {
   }
 
   private validateContext(): CompileError[] {
-    if (this.declarationNode.parent instanceof ProgramNode || !([ElementKind.Table, ElementKind.Project] as (ElementKind | undefined)[]).includes(getElementKind(this.declarationNode.parent).unwrap_or(undefined))) {
+    if (
+      !(this.declarationNode.parent instanceof ProgramNode)
+      && !([ElementKind.Table, ElementKind.Project] as (ElementKind | undefined)[]).includes(getElementKind(this.declarationNode.parent).unwrap_or(undefined))
+    ) {
       return [new CompileError(CompileErrorCode.INVALID_NOTE_CONTEXT, 'A Note can only appear inside a Table or a Project', this.declarationNode)];
     }
 
     return [];
   }
 
   private validateName(nameNode?: SyntaxNode): CompileError[] {
-    if (nameNode) {
-      return [new CompileError(CompileErrorCode.UNEXPECTED_NAME, 'A Note shouldn\'t have a name', nameNode)];
+    if (!(this.declarationNode.parent instanceof ProgramNode)) {
+      if (nameNode) {
+        return [new CompileError(CompileErrorCode.UNEXPECTED_NAME, 'A Note shouldn\'t have a name', nameNode)];
+      }
+      return [];
+    }
+
+    if (!nameNode) {
+      return [new CompileError(CompileErrorCode.INVALID_NAME, 'Sticky note must have a name', this.declarationNode)];
+    }
+
+    const nameFragments = destructureComplexVariable(nameNode);
+    if (!nameFragments.isOk()) return []
```

Comment:
Should this return a CompileError that complain about an invalid name?

Reply:
I have no idea what is not ok em. Can you help provide context for when does names can is not ok so that we can improve the error?

Reply:
`destructureComplexVariable` takes an expression a, if it's of the form `<name>.<name>.<name>` or `<name>` then it's OK. otherwise it's not OK.